libs/features/commerce/src/service.js

```diff
@@ -0,0 +1,126 @@
+import Checkout from './checkout.js';
+import defaults from './defaults.js';
+import { pollImsCountry } from './ims.js';
+import Log from './log.js';
+import getSettings from './settings.js';
+import Wcs from './wcs.js';
+
+const ErrorMessage = {
+  init: 'Not initialised',
+};
+
+/** @type {Commerce.Instance} */
+let instance = null;
+let promise = null;
+const providers = {
+  /** @type {Set<Commerce.providePriceOptions>} */
+  price: new Set(),
+};
+
+async function activate(callback) {
+  const config = callback?.() ?? {};
+
+  Log.init(config.env);
+  const log = Log.commerce;
+  log.debug('Initialising:', config);
+  const literals = { price: {} };
+  const settings = getSettings(config);
+  const imsCountry = pollImsCountry().then((countryCode) => {
+    if (countryCode) {
+      log.debug('Ims country code:', countryCode);
+      settings.country = countryCode;
+    }
+    return countryCode;
+  });
+
+  instance = {
+    checkout: Checkout(settings),
+    defaults: Object.freeze({ ...defaults }),
+    ims: {
+      get country() {
+        return imsCountry;
+      }
+    },
+    providers: {
+      price(provider) {
+        providers.price.add(provider);
+        return () => providers.price.delete(provider);
+      },
+    },
+    literals,
+    settings,
+    wcs: Wcs(settings)
+  };
+
+  // fetch price literals
+  await window.fetch(
+    await import.meta.resolve(`./literals/price/${settings.language.toLowerCase()}.json`),
+  )
+    .then((response) => response.json())
+    .then(
+      (priceLiterals) => {
+        literals.price = priceLiterals;
+        log.debug('Price literals loaded:', priceLiterals);
+      },
+      (error) => {
+        /* c8 ignore next */
+        log.error('Price literals not loaded:', error);
+      },
+    );
+
+  // light up all DOM placeholders
+  document
+    .querySelectorAll('[is="checkout-link"],[is="inline-price"]')
+    // @ts-ignore
+    .forEach((el) => el.render());
```

Comment:
Ideally it should also initialise custom placeholders.
Maybe we should dispatch an event instead?

Reply:
Why shouldn't we?
Attributes `is="checkout-link"` and `is="inline-price"` are introduced exactly in Dexter Commerce and are well known to this module. And it looks natural that just initialised service fires up all placeholders.
Maybe this reduces flexibility on consumer requiring to perform some extra attribution of DOM placeholder elements.
Let's discuss this offline.